pkg/kubelet/kubelet.go

```diff
@@ -1474,7 +1479,44 @@ func (kl *Kubelet) killPod(pod *api.Pod, runningPod *kubecontainer.Pod, status *
 	} else if status != nil {
 		p = kubecontainer.ConvertPodStatusToRunningPod(status)
 	}
-	return kl.containerRuntime.KillPod(pod, p, gracePeriodOverride)
+	// cache the pod cgroup Name for setting the cgroups limits on the pod cgroup later
+	pcm := kl.containerManager.NewPodContainerManager()
+	var podCgroup string
+	reduceMemoryLimts := true
+	if pod != nil {
+		podCgroup = pcm.GetPodContainerName(pod)
+	} else {
+		// If the pod is nil then cgroup limit must have already
+		// been decreased earlier
+		reduceMemoryLimts = false
```

Comment:
This logic is not going to work. Instead spawn a go thread to handle deletion in the background.